File: cpp/stack/stack.cpp
#include <iostream>

//#include <algorithm>

using namespace std;

class sty {


	public: const static int size =5;
	public:  int top ;
	public:  int Array[size];
   //public: vector<int> v;
  public:sty(){
		top =-1;
	}

  public: void push(){
         int value;
       if (top == size-1){
              cout << "overflow" << '\n';
           }
      else{
           cout <<"Enter  a number to push it in array" << endl;
           cin >> value;
           Array[++top]= value;
          }



      }
public: void pop() {
        if (top == -1) {
             cout << "underflow" << '\n';
               }
        else
               {
                 top--;
               }

     }
public: void show() {
           if (top == -1) {
                 cout << "Emapaty" << '\n';
             }
         else{
                for (int i = 0; i < top; i++) {
                 cout << Array[i] << '\t';
                  }

              }
      }
};
int main(int argc, char const *argv[]) {
        sty obj;
        int option;
  // cout << obj.v.capacity() << '\n';
// cout << obj.v.max_size() << endl;
     while (true) {
          cout << "1 \t push" <<endl;
          cout << "2 \t pop" <<endl;
          cout << "3 \t show" <<endl;
          cout << "4 \t Exit"<<endl;
          cin >> option;
         switch (option) {
            case 1:
               obj.push();
            break;
            case 2:
                obj.pop();
            break;
            case 3:
                obj.show();
            break;
            case 4:
              exit(0);
            break;

          }

       }
           return 0;
}

File: python/stack/stack.py
class stack:
  cars="sujki";

  def __init__(self):
        self.nu = []
        self.top =0

        self.max = int(input("What would be size of array "));
        self.end = self.max
  def push(self):
     if self.top < self.max:
          x = int(input("Enter a Number "))
          p.nu.append(x)
          self.top+=1
     else:
          print("overfllow")

  def pop(self):
     if self.top ==0:
        print("underflow")
     else:
         self.end =self.end-1 
  def show(self):
     if self.top > self.max:
         print("overflow")
     else:
         i = 0
         while i < self.end:
             print(self.nu[i])
             i+=1
p=stack()
while True:
   print("1 \t push")
   print("2 \t pop")
   print("3 \t show")
   print("4 \t Exit")
   option = int(input("choose option "))
   if option == 1:
        p.push();
   elif option == 2:
        p.pop();
   elif option == 3:
        p.show()
   elif option == 4:
       break
   else:
       print("faiiled")

               # print(c)
               # for c in p.nu[]:
